### DISCUSSION TITLE
predict_survival_function not working properly

### DISCUSSION THREAD

**Discussion by jiva-kalan** [sebp/scikit-survival#409] [Q&A]
I keep getting this issue when trying to use predict_survival_function 
```python
surv = rsf.predict_survival_function(df)
```
```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Users\username\Miniconda3\envs\churn_model_311\lib\site-packages\sksurv\ensemble\forest.py", line 602, in predict_survival_function
    return super().predict_survival_function(X, return_array)
  File "C:\Usersusername\Miniconda3\envs\churn_model_311\lib\site-packages\sksurv\ensemble\forest.py", line 268, in predict_survival_function
    return _array_to_step_function(self.unique_times_, arr)
AttributeError: 'RandomSurvivalForest' object has no attribute 'unique_times_'
```
Does anyone know why this is happening? My code was working using py3.8 + scikit-survival 0.15.0.post0  and I'm exploring updating python to 3.11 and scikit-survival to 0.21

**Comment by sebp** [comment]
Could you please provide a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) to help me understand the issue.

**Comment by HASAN-Ammar** [comment]
Hi,

I had the same issue, so I solved it by changing my code as follows:

surv = rsf.predict_survival_function(X_test_sel, return_array=False)
i=0
for fn in surv:
    plt.step(fn.x, fn(fn.x), where="post", label=str(i))
    i=i+1
plt.ylabel("Survival probability")
plt.xlabel("Time in days")
plt.legend()
plt.grid(True)

**Reply by HASAN-Ammar** [reply]
Or you can use 
event_times_ instead of unique_times_

surv = rsf.predict_survival_function(X_test_sel, return_array=True)
for i, s in enumerate(surv):
    plt.step(rsf.event_times_, s, where="post", label=str(i))

plt.ylabel("Survival probability")
plt.xlabel("Time in days")
plt.legend()
plt.grid(True)
